### ISSUE TITLE
Does not work correct with negative numbers

### ISSUE THREAD

**Issue by tilkinsc** [frogstair/number-system-converter#4]
![image](https://github.com/frogstair/number-system-converter/assets/7494772/5ad94332-3e95-40df-bf53-7902e9d53ebc)

If you highlight a number and change type, it appears to preserve the negative sign, but the numbers seem to be incorrect as well. Correct implementation would be to remove the sign and generate the correct number. I think a drawback here is that the type can't be easily inferred, so I think the command palette could let you choose the size.

**Comment by crabtutorial** [comment]
I dont think thats viable to implement because you dont know the size of the number. How do I know if the number is 8, 16 or 32 bit given a base 10 integer?

**Comment by tilkinsc** [comment]
Dont you have access to the text users are interacting with?

**Comment by crabtutorial** [comment]
Yes, but not the context. The number might be 32 bits in the code being written, but be converted to 8 bits, and that will cause errors

C example:

`int32_t value = -10;`

Gets converted to

`int32_t value = 0b11110110`

or

`int32_t value = 0xF6`

Which is -10 in binary/hex, but not the correct amount of bits, and the hex representation is entirely wrong

**Comment by tilkinsc** [comment]
As it currently stands the implementation is broken. Further, unsupported. So something is better than nothing.

In the literal sense 0xF6 is wrong for the type, but its right for the value.

On one route you could focus on the amount of bytes in the number and take the user literally. Users attempting to use hex are more than capable of padding extra 1's or 0's or F's to achieve the desired length, if necessary. if not(X)+1 >= N where X is input, N is 0xFF or 0xFFFFFFFF should be used in this circumstance to more accurately guess the intended size.

signed -128
unsigned 128
hex 0x80

signed -32768
unsigned 32768
hex 0x8000

Another route, you could make a drop down from the palette to select the bit length just for negative numbers.

The funny route, you scan the line for any mention of keywords uint int byte ubyte "unsigned byte" "signed byte" etc and choose the closest one to the left of the start of the index of the number string. By default any time you do 10 or -10 freely you can assume its an int.
